src/functions/severa/get-resource-allocations-by-user/handler.ts

```diff
@@ -0,0 +1,62 @@
+import type { APIGatewayProxyHandler } from "aws-lambda";
+import { CreateSeveraApiService } from "src/database/services/severa-api-service";
+import { middyfy } from "src/libs/lambda";
+import type ResourceAllocationModel from "src/types/severa/resourceAllocation/resourceAllocation";
+import type SeveraResponseResourceAllocation from "src/types/severa/resourceAllocation/severaResponseResourceAllocation";
+
+/**
+ * Handler for getting resourceAllocation by user from Severa REST API.
+ * 
+ * @param event - API Gateway event containing the userId.
+ */
+export const getResourceAllocationHandler: APIGatewayProxyHandler = async (event) => {
+const {severaUserId} = event.pathParameters
+
+try {
+    const api = CreateSeveraApiService();
+
+    if(!severaUserId) {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({ message: "User Id is required" }),
+      };
+    }
```

Comment:
Do validation at the earliest point possible, I'd move this straight after destructuring the severaUserId